packages/datadog-plugin-iovalkey/test/index.spec.js

```diff
@@ -0,0 +1,194 @@
+'use strict'
+
+const agent = require('../../dd-trace/test/plugins/agent')
+const { breakThen, unbreakThen } = require('../../dd-trace/test/plugins/helpers')
+const { ERROR_MESSAGE, ERROR_TYPE, ERROR_STACK } = require('../../dd-trace/src/constants')
+
+const { expectedSchema, rawExpectedSchema } = require('./naming')
+
+describe('Plugin', () => {
+  let Valkey
+  let valkey
+  let tracer
+
+  describe('iovalkey', () => {
+    withVersions('iovalkey', 'iovalkey', version => {
+      beforeEach(() => {
+        tracer = require('../../dd-trace')
+        Valkey = require(`../../../versions/iovalkey@${version}`).get()
+        valkey = new Valkey({ connectionName: 'test' })
+      })
+
+      afterEach(() => {
+        unbreakThen(Promise.prototype)
+        valkey.quit()
+      })
+
+      describe('without configuration', () => {
+        before(() => agent.load(['iovalkey']))
+
+        after(() => agent.close({ ritmReset: false }))
+
+        it('should do automatic instrumentation when using callbacks', done => {
+          agent.use(() => {}) // wait for initial info command
+          const runs = ['info', 'client', 'get']
+          let run = 0
+          agent.use(traces => {
+            const expectedResource = runs[run++]
+            expect(traces[0][0]).to.have.property('resource', expectedResource)
+            if (expectedResource !== 'client') return
+
+            expect(traces[0][0]).to.have.property('name', expectedSchema.outbound.opName)
+            expect(traces[0][0]).to.have.property('service', expectedSchema.outbound.serviceName)
+            expect(traces[0][0]).to.have.property('resource', 'get')
+            expect(traces[0][0]).to.have.property('type', 'redis')
```

Comment:
I think the last important change is that type and db.type should be set to valkey instead of redis.